frontend/plugins/acm/resource-routes.ts

```diff
@@ -0,0 +1,63 @@
+/* Copyright Contributors to the Open Cluster Management project */
+
+import { generatePath } from 'react-router-dom-v5-compat'
+import queryString from 'query-string'
+import { ResourceRouteHandler } from '@stolostron/multicluster-sdk'
+
+/**
+ * ACM resource route handler for first-class resource pages.
+ * Routes resources to their dedicated ACM pages instead of generic search results.
+ */
+export const acmResourceRouteHandler: ResourceRouteHandler = ({ cluster, namespace, name, resource, model }) => {
+  const { kind, group } = model
+
+  switch (kind.toLowerCase()) {
+    case 'cluster':
+    case 'managedcluster':
+      return generatePath('/multicloud/infrastructure/clusters/details/:namespace/:name/overview', {
```

Comment:
Let's move this file somewhere under frontend/src and use the NavigationPath constants, as the original code does.